cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -1,44 +1,96 @@
 
 .. _logging-monitoring:
 
 ======================
 Logging and Monitoring
 ======================
 
-Logs and Metrics
-================
-CDAP collects logs and metrics for all of its internal services and and user applications.
+CDAP collects logs and metrics for all of its internal services and on user applications.
 Being able to view these details can be very helpful in debugging CDAP applications as
 well as analyzing their performance. CDAP gives access to its logs, metrics, and other
-monitoring information through RESTful APIs as well as a Java Client.
+monitoring information through RESTful APIs, the CDAP UI, as well as a Java Client.
 
-See the :ref:`Logging <http-restful-api-logging>`, :ref:`Metrics <http-restful-api-metrics>`, 
-and :ref:`Monitoring <http-restful-api-monitor>` HTTP RESTful APIs, the :ref:`Java Client
-<reference:java-client-api>`, and the :ref:`master-service-logging-configuration` for
-additional information.
+In Hadoop clusters, the programs running in containers generate individual log files as
+part of the container. As these files generally do not persist beyond the life of the
+container, they are volatile and not very useful for post-mortem diagnostics and
+troubleshooting, or for analyzing performance. As a single application can consist of
+multiple  programs distributed across the nodes of a cluster, the complete logs for the
+application may similarly be distributed.
 
+To address these issues, the CDAP log framework was designed:
 
-Log Location
-============
-The location of logs depends on the mode of CDAP and the Hadoop distribution.
+- to centralize the location of logs, so that the individual logs of an application's programs can be
+  merged into one;
 
-- With **Standalone CDAP** (the CDAP SDK), logs are located in ``<CDAP-SDK-HOME>/logs``.
+- to make logs persistent and available for later use and analysis;
 
-- For **Distributed CDAP**, with the exception of Cloudera Manager, logs are located in
-  ``/var/log/cdap``. With Cloudera Manager installations, log files are located in
-  directories under ``/var/run/cloudera-scm-agent/process``.
+- to be extensible using :ref:`custom log pipelines <logging-monitoring-custom-logging-pipeline>`; and
 
+- to use a system that can be adapted for individual requirements at both the application
+  and cluster levels.
 
-Logging Messages
-================
 
+Logging Example
+===============
+.. highlight:: console
+
+This diagram shows the steps CDAP follows when logging a program of an application:
+
+  - Logs are collected from an individual program running in a YARN container.
+
+  - YARN containers have their own local log files.
+  
+  - CDAP logs are published to Kafka.
+
+  - The Log Saver Service (*Log.saver*) is configured to read log messages from Kafka.
+    Log.saver reads the messages from Kafka, buffers and sorts them in-memory, before
+    persisting them to HDFS. These messages are written to files corresponding to the
```

Comment:
Each of these files corresponds to one program or application, depending on how the grouping is configured. (see reference)